forms/HtmlEditorField.php

```diff
@@ -518,8 +518,12 @@ public function getanchors() {
 			}
 
 			// Similar to the regex found in HtmlEditorField.js / getAnchors method.
-			if (preg_match_all("/name=\"([^\"]+?)\"|name='([^']+?)'/im", $page->Content, $matches)) {
-				$anchors = $matches[1];
+			if (preg_match_all(
+				"/\s+(name|id)\s*=\s*([\"'])([^\\2\s>]*?)\\2|\s+(name|id)\s*=\s*([^\"']+)[\s +>]/im",
```

Comment:
The 2 refers to the second match (that comes from `([\"'])`). It’s used to match any string in the attribute value that isn’t the opening quote. Basically it allows you to have `<div name="single'quote">` or `<div name='double"quote'>`, rather than having to write two separate rules for each type of quote mark